lib/hexo/post.js

```diff
@@ -28,30 +25,153 @@ const _restoreContent = cache => (_, index) => {
   return value;
 };
 
-class PostRenderCache {
+const STATE_PLAINTEXT = Symbol('plaintext');
+const STATE_SWIG_VAR = Symbol('swig_var');
+const STATE_SWIG_COMMENT = Symbol('swig_comment');
+const STATE_SWIG_TAG = Symbol('swig_tag');
+const STATE_SWIG_FULL_TAG = Symbol('swig_full_tag');
+
+const isNonWhiteSpaceChar = char => char !== '\r'
+  && char !== '\n'
+  && char !== '\t'
+  && char !== '\f'
+  && char !== '\v'
+  && char !== ' ';
+
+class PostRenderEscape {
   constructor() {
-    this.cache = [];
+    this.stored = [];
   }
 
   restoreAllSwigTags(str) {
-    const restored = str.replace(rSwigPlaceHolder, _restoreContent(this.cache));
-    if (restored === str) return restored;
-    return this.restoreAllSwigTags(restored); // self-recursive for nexted escaping
+    const restored = str.replace(rSwigPlaceHolder, _restoreContent(this.stored));
+    return restored;
```

Comment:
The self-recursive escaping is firstly introduced by @seaoak in #2321 trying to mitigate a `back_tick_code` inside a `swig tag` issue when the year is 2016. Since then, Hexo's post rendering mechanics has changed dramatically, and when @2321 being merged in 2019, there is no longer required self-recursive as nested escaping will never happen. After this change, the unit test case still passed.